morebits.js

```diff
@@ -1308,29 +1308,239 @@ Morebits.unbinder.getCallback = function UnbinderGetCallback(self) {
-	'July', 'August', 'September', 'October', 'November', 'December' ];
+	this._d = new (Function.prototype.bind.apply(Date, [Date].concat(args)));
 
-Date.monthNamesAbbrev = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+	if (isNaN(this._d.getTime()) && typeof args[0] === 'string') {
+		// Try again after removing a comma, to get MediaWiki timestamps to parse
+		this._d = new (Function.prototype.bind.call(Date, Date, args[0].replace(/(\d\d:\d\d),/, '$1')));
+	}
 
-Date.prototype.getUTCMonthName = function() {
-	return Date.monthNames[this.getUTCMonth()];
 };
 
-Date.prototype.getUTCMonthNameAbbrev = function() {
-	return Date.monthNamesAbbrev[this.getUTCMonth()];
+Morebits.date.localeData = {
+	months: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
+	monthsShort: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
+	days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+	daysShort: ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'],
+	relativeTimes: {
+		thisDay: '[Today at] h:mm A',
+		prevDay: '[Yesterday at] h:mm A',
+		nextDay: '[Tomorrow at] h:mm A',
+		thisWeek: 'dddd [at] h:mm A',
+		lastWeek: '[Last] dddd [at] h:mm A',
+		other: 'DD/MM/YYYY'
+	}
 };
 
+// Allow native Date.prototype methods to be used on Morebits.date objects
+Object.getOwnPropertyNames(Date.prototype).forEach(function(func) {
+	Morebits.date.prototype[func] = function() {
+		return this._d[func].apply(this._d, Array.prototype.slice.call(arguments));
+	};
+});
+
+$.extend(Morebits.date.prototype, {
+
+	isValid: function() {
+		return !isNaN(this.getTime());
+	},
+
+	/** @param {(Date|Morebits.date)} date */
+	isBefore: function(date) {
+		return this.getTime() < date.getTime();
+	},
+	isAfter: function(date) {
+		return this.getTime() > date.getTime();
+	},
+
+	/** @return {string} */
+	getUTCMonthName: function() {
+		return Morebits.date.localeData.months[this.getUTCMonth() + 1];
+	},
+	getUTCMonthNameAbbrev: function() {
+		return Morebits.date.localeData.monthsShort[this.getUTCMonth() + 1];
+	},
+	getMonthName: function() {
+		return Morebits.date.localeData.months[this.getMonth() + 1];
+	},
+	getMonthNameAbbrev: function() {
+		return Morebits.date.localeData.monthsShort[this.getMonth() + 1];
+	},
+	getUTCDayName: function() {
+		return Morebits.date.localeData.days[this.getUTCDay()];
+	},
+	getUTCDayNameAbbrev: function() {
+		return Morebits.date.localeData.daysShort[this.getUTCDay()];
+	},
+	getDayName: function() {
+		return Morebits.date.localeData.days[this.getDay()];
+	},
+	getDayNameAbbrev: function() {
+		return Morebits.date.localeData.daysShort[this.getDay()];
+	},
+
+	/**
+	 * Add a given number of minutes, hours, days, months or years to the date.
+	 * This is done in-place. The modified date object is also returned, allowing chaining.
+	 * @param {number} number
```

Comment:
Added the comment. 

According to the ECMAScript specification[0], Date.set* methods use floor. So we don't need to worry about it at our end. Though we could use `Math.round` which could be considered better than `floor`. But I think it's best to avoid such unnecessary "improvements". 

[0]: https://tc39.es/ecma262/#sec-date.prototype.setdate, https://tc39.es/ecma262/#sec-makeday, https://tc39.es/ecma262/#sec-tointeger